src/pygeoprocessing/geoprocessing.py

```diff
@@ -2722,10 +2732,23 @@ def transform_bounding_box(
     target_ref = osr.SpatialReference()
     target_ref.ImportFromWkt(target_ref_wkt)
 
-    transformer = osr.CoordinateTransformation(base_ref, target_ref)
+    # Create a coordinate transformation [GDAL 2 NOT supported]
+    # In GDAL 3.0 spatial references have been updated. One difference is 
+    # how transforms work with lat/lon coordinate systems. Transforms
+    # expect (lat, lon) order in GDAL 3.0 as opposed to thge GIS friendly
+    # (lon, lat). Thus if using a (lat, lon) srs set for backwards 
+    # compatability. 
+    if not base_ref.IsProjected():
+        base_ref.SetAxisMappingStrategy(osr.OAMS_TRADITIONAL_GIS_ORDER)
+    if not target_ref.IsProjected():
+        target_ref.SetAxisMappingStrategy(osr.OAMS_TRADITIONAL_GIS_ORDER)
```

Comment:
Reading through https://trac.osgeo.org/gdal/wiki/rfc73_proj6_wkt2_srsbarn#Axisorderissues, I think the test for not [...].IsProjected(): might not be necessary, and in some cases wrong. Like some kind of exotic projected transformation that swaps its coordinates. From what I understand in the case here where we're always assuming x/y order, we should assume the osr.OAMS_TRADITIONAL_GIS_ORDER for both transforms. Is that right?

Reply:
Hey Rich, I just spent a significant amount of time in the WKT2 docs, PROJ 6 docs, and GDAL 3 Docs. I _think_ I'm starting to develop a deeper understanding. I think the basics are as follows:

- GDAL 3 conforms to the authority defining coordinate reference systems (CRS). 
- For Geographic CRS (WGS84), this means no longer ignoring Axis Order but abiding by LAT,LON order
- For Projected CRS (UTM), most EPSG in the registry use Easting, Northing order
- We think of things in general as **x**,**y** because it is convenient, but now a `Transform` expects Axis Order, where  if `long=x` and `lat=y`, GDAL 3 would take `Transform(y, x)`
- Thus when transforming from Projected CRS to Geographic CRS, `Transform(x,y)` will return `y,x` (Lat,Lon)
- Thus when transforming from Geographic CRS to Projected CRS, `Transform(y,x)` will return `x,y` (Easting, Northing)
- Thus when transforming from Geographic CRS to Geographic CRS, `Transform(y,x)` will return `y,x` (Lat,Lon)
- Thus when transforming from Projected CRS to Projected CRS, `Transform(x,y)` will return `x,y` (Easting, Northing)

**note:** Most projected CRS use Easting first, Northing second convention, however, some defined in the EPSG registry use the reverse convention.

So, I think you're correct in that `not.____.IsProjected()` is probably wrong, but `_____.IsGeographic()` _could_ be OK. Basically that Axis Order setting `SetAxisMappingStrategy(osr.OAMS_TRADITIONAL_GIS_ORDER)` is saying, if we've got a Geographic CRS, use Lon,Lat like the good old days (since we think of `lon,lat` as `x,y`. We certainly can, but maybe should, **NOT** do that. 

Perhaps a better solution is to create a mapping of Axis Order to Geographic (Lat,Lon) and Projected (Easting, Northing), such that when handling the extents and bounding box points we can correctly return _x_, _y_.

Reply:
I think it's always okay to use `srs.SetAxisMappingStrategy(osr.OAMS_TRADITIONAL_GIS_ORDER)` in this case. It'll only "swap" x/y if it's in a coordinate system that needs it. UTM projections are not affected. At least that's how I read the docs, and also I tried it manually and it behaved as expected. Is that okay?

Reply:
Yea, I think that's right!